modules/s3/main.tf

```diff
@@ -0,0 +1,33 @@
+# tfsec:ignore:aws-s3-encryption-customer-key tfsec:ignore:aws-s3-enable-bucket-encryption
+resource "aws_s3_bucket" "this" {
+  #checkov:skip=CKV2_AWS_61: There is no need to add lifecycle configuration to S3 bucket
+  #checkov:skip=CKV2_AWS_62: There is no need to add event notification to S3 bucket
+  #checkov:skip=CKV_AWS_145: There is no need to encrypt S3 bucket with KMS
+  #checkov:skip=CKV_AWS_144: There is no need to replicate S3 bucket
+  bucket        = local.bucket_name
+  force_destroy = local.force_destroy
+}
+
+resource "aws_s3_bucket_public_access_block" "this" {
+  bucket = aws_s3_bucket.this.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
```

Comment:
the ticket says configure s3 bucket for app assets, how app assets will be loaded if there is no public access

are you going to configure CDN on top of that?

or assets are not css and js files, but something else?

Reply:
I think it's good enough at the moment since we didn't integrate ECS yet. 

I think we can customize it later (after deploying the web application) or only accept the internal request (from ECS only), and the ECS will serve these assets to the user, what do you think? 💭

Reply:
it means that you will deploy application to ECS that will proxy assets requests to S3? usually it is not the way how you want to distribute assets for the app

usually there is CDN on top of S3 or there is application server (nginx) that can effectively stream assets to the user

I'm just trying to understand how this S3 will be used

Reply:
or this bucket is not for css and js? maybe it is for files that will be uploaded by user?

Reply:
Ah, I got your question. I intend to serve only the files uploaded by the user as you mentioned and serve the CSS/JS directly from the application, should we do that way?

Reply:
yep, it is ok, in that case it shouldn't be public, "assets" word just confused me

thanks!